src/main/java/FIFOSchedular.java

```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class FIFOSchedular implements TaskSchedular {
+
+    List<Task> taskList;
+    Map<Integer, HashSet<UUID>> taskGroupsBasedOnPriority;
```

Comment:
Can we extract HashSet<UUID> in another model? That way u can keep related behaviour along with it.